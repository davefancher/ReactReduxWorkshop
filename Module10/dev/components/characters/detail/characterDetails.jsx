import React, { Component } from "react";
import Loading from "../../shared/loading.jsx";
import LifeEventField from "../lifeEventField.jsx";

const List = (props) => {
    if (props.length === 0) return null;

    const labelStyle = {
        marginRight: "3px",
        marginBottom: "3px",
        fontSize: "14px",
        display: "inline-block",
        fontWeight: "normal"
    };

    return (
        <div>
            {props.map((p, ix) => <span key={ix} style={labelStyle} className="label label-default">{p}</span>)}
        </div>
    )
}
const PropertyGroup = (props) => {
    var content = null;

    if(props.children) {
        content = props.children;
    } else if (!props.value) {
        return null;
    } else if(Array.isArray(props.value)) {
        content = List(props.value);
    } else if (props.value.location && props.value.date) {
        content = <div>{props.value.date}<br />{props.value.location}</div>;
    } else if (props.value.location) {
        content = props.value.location;
    } else if (props.value.date) {
        content = props.value.date;
    } else {
        content = props.value;
    }

    return (
        <div>
            <dt>{props.title}:</dt>
            <dd>{content || <i className="text-muted">Unknown</i>}</dd>
        </div>
    );
};

export class CharacterDetails extends Component {
    componentDidMount () {
        this.props.getCharacter();
    }

    render () {
        if (this.props.loading) {
           return <Loading />;
        }

        var character = this.props.character;

        return (
            <div>
                <h2>{character.name ? character.name : <i>{character.aliases[0]}*</i>}</h2>
                <dl className="dl-horizontal">
                    <PropertyGroup title="Gender" value={character.gender} />
                    <PropertyGroup title="Culture" value={character.culture} />
                    <PropertyGroup title="Born">
                        <LifeEventField date={character.birthDate}
                                        location={character.birthLocation} />
                    </PropertyGroup>
                    <PropertyGroup title="Died">
                        <LifeEventField date={character.deathDate}
                                        location={character.deathLocation} />
                    </PropertyGroup>
                    <PropertyGroup title="Aliases" value={character.aliases} />
                    <PropertyGroup title="Titles" value={character.titles} />
                    <PropertyGroup title="Portrayed By" value={character.portrayedBy} />
                </dl>

                {character.name ? "" : <p className="pull-right"><small className="text-muted">* - Name unknown, using alias instead</small></p>}
            </div>
        );
    }
}
