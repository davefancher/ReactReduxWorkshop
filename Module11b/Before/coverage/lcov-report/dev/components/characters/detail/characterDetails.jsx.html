<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for dev/components/characters/detail/characterDetails.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">dev/components/characters/detail</a> characterDetails.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">18.52% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/27</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">12.5% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>3/24</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">20% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">19.23% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/26</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from "react";
import Loading from "../../shared/loading.jsx";
import LifeEventField from "../lifeEventField.jsx";
&nbsp;
export const List = <span class="fstat-no" title="function not covered" >(p</span>rops) =&gt; {
<span class="cstat-no" title="statement not covered" >    if (props.length === 0) <span class="cstat-no" title="statement not covered" >return null;</span></span>
&nbsp;
    const labelStyle = <span class="cstat-no" title="statement not covered" >{</span>
        marginRight: "3px",
        marginBottom: "3px",
        fontSize: "14px",
        display: "inline-block",
        fontWeight: "normal"
    };
&nbsp;
<span class="cstat-no" title="statement not covered" >    return (</span>
        &lt;div&gt;
            {props.map(<span class="fstat-no" title="function not covered" >(p</span>, ix) =&gt; <span class="cstat-no" title="statement not covered" >&lt;span key={ix} style={labelStyle} className="label label-default"&gt;{p}&lt;/span&gt;)</span>}
        &lt;/div&gt;
    )
}
export const PropertyGroup = <span class="fstat-no" title="function not covered" >(p</span>rops) =&gt; {
    var content = <span class="cstat-no" title="statement not covered" >null;</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >    if(props.children) {</span>
<span class="cstat-no" title="statement not covered" >        content = props.children;</span>
    } else <span class="cstat-no" title="statement not covered" >if (!props.value) {</span>
<span class="cstat-no" title="statement not covered" >        return null;</span>
    } else <span class="cstat-no" title="statement not covered" >if(Array.isArray(props.value)) {</span>
<span class="cstat-no" title="statement not covered" >        content = List(props.value);</span>
    } else <span class="cstat-no" title="statement not covered" >if (props.value.location &amp;&amp; props.value.date) {</span>
<span class="cstat-no" title="statement not covered" >        content = &lt;div&gt;{props.value.date}&lt;br /&gt;{props.value.location}&lt;/div&gt;;</span>
    } else <span class="cstat-no" title="statement not covered" >if (props.value.location) {</span>
<span class="cstat-no" title="statement not covered" >        content = props.value.location;</span>
    } else <span class="cstat-no" title="statement not covered" >if (props.value.date) {</span>
<span class="cstat-no" title="statement not covered" >        content = props.value.date;</span>
    } else {
<span class="cstat-no" title="statement not covered" >        content = props.value;</span>
    }
&nbsp;
<span class="cstat-no" title="statement not covered" >    return (</span>
        &lt;div&gt;
            &lt;dt&gt;{props.title}:&lt;/dt&gt;
            &lt;dd&gt;{content || &lt;i className="text-muted"&gt;Unknown&lt;/i&gt;}&lt;/dd&gt;
        &lt;/div&gt;
    );
};
&nbsp;
export class CharacterDetails extends Component {
<span class="fstat-no" title="function not covered" >    co</span>mponentDidMount () {
<span class="cstat-no" title="statement not covered" >        this.props.getCharacter();</span>
    }
&nbsp;
    render () {
        <span class="missing-if-branch" title="if path not taken" >I</span>if (this.props.loading) {
<span class="cstat-no" title="statement not covered" >           return &lt;Loading /&gt;;</span>
        }
&nbsp;
        var character = this.props.character;
&nbsp;
        return (
            &lt;div&gt;
                &lt;h2&gt;{character.name ? character.name : <span class="branch-1 cbranch-no" title="branch not covered" >&lt;i&gt;{character.aliases[0]}*&lt;/i&gt;}</span>&lt;/h2&gt;
                &lt;dl className="dl-horizontal"&gt;
                    &lt;PropertyGroup title="Gender" value={character.gender} /&gt;
                    &lt;PropertyGroup title="Culture" value={character.culture} /&gt;
                    &lt;PropertyGroup title="Born"&gt;
                        &lt;LifeEventField date={character.birthDate}
                                        location={character.birthLocation} /&gt;
                    &lt;/PropertyGroup&gt;
                    &lt;PropertyGroup title="Died"&gt;
                        &lt;LifeEventField date={character.deathDate}
                                        location={character.deathLocation} /&gt;
                    &lt;/PropertyGroup&gt;
                    &lt;PropertyGroup title="Aliases" value={character.aliases} /&gt;
                    &lt;PropertyGroup title="Titles" value={character.titles} /&gt;
                    &lt;PropertyGroup title="Portrayed By" value={character.portrayedBy} /&gt;
                &lt;/dl&gt;
&nbsp;
                {character.name ? "" : <span class="branch-1 cbranch-no" title="branch not covered" >&lt;p className="pull-right"&gt;&lt;small className="text-muted"&gt;* - Name unknown, using alias instead&lt;/small&gt;&lt;/p&gt;}</span>
            &lt;/div&gt;
        );
    }
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Sep 27 2017 13:57:49 GMT-0400 (EDT)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
