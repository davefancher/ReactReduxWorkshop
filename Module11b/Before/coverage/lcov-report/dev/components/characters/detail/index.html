<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for dev/components/characters/detail</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> dev/components/characters/detail
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">18.52% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/27</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">12.5% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>3/24</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">20% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">19.23% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/26</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="characterDetails.jsx"><a href="characterDetails.jsx.html">characterDetails.jsx</a></td>
	<td data-value="18.52" class="pic low"><div class="chart"><div class="cover-fill" style="width: 18%;"></div><div class="cover-empty" style="width:82%;"></div></div></td>
	<td data-value="18.52" class="pct low">18.52%</td>
	<td data-value="27" class="abs low">5/27</td>
	<td data-value="12.5" class="pct low">12.5%</td>
	<td data-value="24" class="abs low">3/24</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="5" class="abs low">1/5</td>
	<td data-value="19.23" class="pct low">19.23%</td>
	<td data-value="26" class="abs low">5/26</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Sep 27 2017 13:57:49 GMT-0400 (EDT)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
