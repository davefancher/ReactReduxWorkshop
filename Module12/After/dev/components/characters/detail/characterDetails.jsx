import React, { Component } from "react";
import { Field, FieldArray, reduxForm, formValueSelector } from 'redux-form';
var ReactToastr = require("react-toastr");
var { ToastContainer } = ReactToastr;
var ToastMessageFactory = React.createFactory(ReactToastr.ToastMessage.animation);
import { connect } from 'react-redux';
import FontAwesome from "react-fontawesome";
import Loading from "../../shared/loading.jsx";
import LifeEventField from "../lifeEventField.jsx";
import IceAndFireRepository from "../../../IceAndFireRepository.js";
import renderInput from '../form/renderInput';
import renderAliasArray from '../form/renderAliasArray';
import validate from '../form/validate';

const List = (props) => {
    if (props.length === 0) return null;

    const labelStyle = {
        marginRight: "3px",
        marginBottom: "3px",
        fontSize: "14px",
        display: "inline-block",
        fontWeight: "normal"
    };

    return (
        <div>
            {props.map((p, ix) => <span key={ix} style={labelStyle} className="label label-default">{p}</span>)}
        </div>
    )
}
const PropertyGroup = (props) => {
    var content = null;

    if(props.children) {
        content = props.children;
    } else if (!props.value) {
        return null;
    } else if(Array.isArray(props.value)) {
        content = List(props.value);
    } else if (props.value.location && props.value.date) {
        content = <div>{props.value.date}<br />{props.value.location}</div>;
    } else if (props.value.location) {
        content = props.value.location;
    } else if (props.value.date) {
        content = props.value.date;
    } else {
        content = props.value;
    }

    return (
        <div>
            <dt>{props.title}:</dt>
            <dd>{content || <i className="text-muted">Unknown</i>}</dd>
        </div>
    );
};

class CharacterDetails extends Component {
    constructor () {
        super();
        this.state = {
            contentEditable: false
        };
        this.toggleEdit = this.toggleEdit.bind(this);
        this.submitForm = this.submitForm.bind(this);
        this.notifySave = this.notifySave.bind(this);
    }

    toggleEdit () {
        this.setState({
            contentEditable: !this.state.contentEditable
        });
    }

    submitForm (e) {
        const character = this.props.character;
        const { gender, culture, aliases } = this.props.formFields.values;
        character.gender = gender;
        character.culture = culture;
        character.aliases = aliases;

        IceAndFireRepository
            .characters
            .store(character)
            .then(character => {
                this.toggleEdit();
                this.notifySave();
            });
    }

    notifySave () {
        this.container.success(
            "Success!",
            "Your character saved.", {
            timeOut: 5000,
            extendedTimeOut: 10000
        });
    }

    componentDidMount () {
        this.props.getCharacter();
    }

    render () {
        if (this.props.loading) {
           return <Loading />;
        }

        var character = this.props.character;

        return (
            <div>
                <header style={{ display: 'flex', justifyContent: 'space-between', alignItems: 'center', margin: '12px' }}>
                    <h2 style={{ margin: 0 }}>
                        {character.name ? character.name : <i>{character.aliases[0]}*</i>}
                    </h2>
                    <span onClick={this.toggleEdit} style={{ cursor: 'pointer' }}>
                        Edit
                        <FontAwesome
                            name='edit'
                            style={{ marginLeft: '7px' }}
                        />
                    </span>
                </header>

                <form onSubmit={this.props.handleSubmit(this.submitForm)}>
                {
                    this.state.contentEditable ?
                        <dl className="dl-horizontal">
                            <Field
                                title="Gender"
                                name="gender"
                                component={renderInput}
                            />
                            <Field
                                title="Culture"
                                name="culture"
                                component={renderInput}
                            />
                            <Field
                                title="Born"
                                name="born"
                                component={renderInput}
                            />
                            <Field
                                title="Died"
                                name="died"
                                component={renderInput}
                            />
                            <FieldArray
                                title="Alias"
                                name="aliases"
                                component={renderAliasArray}
                            />
                            <Field
                                title="Titles"
                                name="titles"
                                component={renderInput}
                            />
                            <Field
                                title="Portrayed By"
                                name="portrayed-by"
                                component={renderInput}
                            />
                            <dt></dt>
                            <dd><button className="btn btn-primary">Save</button></dd>
                        </dl>  :
                        <dl className="dl-horizontal">
                            <PropertyGroup title="Gender" value={character.gender} />
                            <PropertyGroup title="Culture" value={character.culture} />
                            <PropertyGroup title="Born">
                                <LifeEventField date={character.birthDate}
                                                location={character.birthLocation} />
                            </PropertyGroup>
                            <PropertyGroup title="Died">
                                <LifeEventField date={character.deathDate}
                                                location={character.deathLocation} />
                            </PropertyGroup>
                            <PropertyGroup title="Aliases" value={character.aliases} />
                            <PropertyGroup title="Titles" value={character.titles} />
                            <PropertyGroup title="Portrayed By" value={character.portrayedBy} />
                        </dl>
                }
                </form>

                {character.name ? "" : <p className="pull-right"><small className="text-muted">* - Name unknown, using alias instead</small></p>}

                <ToastContainer ref={(input) => {this.container = input;}}
                    toastMessageFactory={ToastMessageFactory}
                    className="toast-top-right"
                />
            </div>
        );
    }
}


let CharacterDetailsForm = reduxForm({
    form: 'characterForm',
    enableReinitialize: true,
    validate
  })(CharacterDetails);

CharacterDetailsForm = connect(
  (state, props) => {

    let initialValues = {};

    const {
        aliases,
        portrayedBy,
        titles,
        birthDate,
        culture,
        deathDate,
        gender,
        name
    } = state.characters.character;

    initialValues = {
        aliases,
        "portrayed-by": portrayedBy,
        titles,
        born: birthDate,
        culture,
        died: deathDate,
        gender,
        name: name || "Unknown"
    };

    return {
        initialValues,
        formFields: state.form.characterForm
    };
  }
)(CharacterDetailsForm);

export default CharacterDetailsForm;
