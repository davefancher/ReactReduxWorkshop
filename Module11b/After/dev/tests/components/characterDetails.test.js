import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import { CharacterDetails, PropertyGroup } from '../../components/characters/detail/characterDetails';
import LifeEventField from "../../components/characters/lifeEventField";

const character = {
    name: "Gwen"
};

test('characterDetails.js renders correctly', () => {
    const component = shallow(
        <CharacterDetails character={character} />
    );
    expect(component).toMatchSnapshot();
})

test('should render seven <PropertyGroup /> components', () => {
    const component = shallow(<CharacterDetails character={character} />);
    expect(component.find(PropertyGroup)).toHaveLength(7);
});

test('should render seven <LifeEventField /> components', () => {
    const component = shallow(<CharacterDetails character={character} />);
    expect(component.find(LifeEventField)).toHaveLength(2);
});

test('should render .dl-horizontal class', () => {
    const component = shallow(<CharacterDetails character={character} />);
    expect(component.find('.dl-horizontal')).toHaveLength(1);
});


