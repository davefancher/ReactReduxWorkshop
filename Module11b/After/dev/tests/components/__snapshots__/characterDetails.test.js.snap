// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`characterDetails.js renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <h2>
      Gwen
    </h2>
    <dl
      className="dl-horizontal"
    >
      <PropertyGroup
        title="Gender"
        value={undefined}
      />
      <PropertyGroup
        title="Culture"
        value={undefined}
      />
      <PropertyGroup
        title="Born"
      >
        <LifeEventField
          date={undefined}
          location={undefined}
        />
      </PropertyGroup>
      <PropertyGroup
        title="Died"
      >
        <LifeEventField
          date={undefined}
          location={undefined}
        />
      </PropertyGroup>
      <PropertyGroup
        title="Aliases"
        value={undefined}
      />
      <PropertyGroup
        title="Titles"
        value={undefined}
      />
      <PropertyGroup
        title="Portrayed By"
        value={undefined}
      />
    </dl>
    
  </div>,
  "nodes": Array [
    <div>
      <h2>
        Gwen
      </h2>
      <dl
        className="dl-horizontal"
      >
        <PropertyGroup
          title="Gender"
          value={undefined}
        />
        <PropertyGroup
          title="Culture"
          value={undefined}
        />
        <PropertyGroup
          title="Born"
        >
          <LifeEventField
            date={undefined}
            location={undefined}
          />
        </PropertyGroup>
        <PropertyGroup
          title="Died"
        >
          <LifeEventField
            date={undefined}
            location={undefined}
          />
        </PropertyGroup>
        <PropertyGroup
          title="Aliases"
          value={undefined}
        />
        <PropertyGroup
          title="Titles"
          value={undefined}
        />
        <PropertyGroup
          title="Portrayed By"
          value={undefined}
        />
      </dl>
      
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <CharacterDetails
        character={
          Object {
            "name": "Gwen",
          }
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": CharacterDetails {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "character": Object {
            "name": "Gwen",
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <h2>
            Gwen
          </h2>
          <dl
            className="dl-horizontal"
          >
            <PropertyGroup
              title="Gender"
              value={undefined}
            />
            <PropertyGroup
              title="Culture"
              value={undefined}
            />
            <PropertyGroup
              title="Born"
            >
              <LifeEventField
                date={undefined}
                location={undefined}
              />
            </PropertyGroup>
            <PropertyGroup
              title="Died"
            >
              <LifeEventField
                date={undefined}
                location={undefined}
              />
            </PropertyGroup>
            <PropertyGroup
              title="Aliases"
              value={undefined}
            />
            <PropertyGroup
              title="Titles"
              value={undefined}
            />
            <PropertyGroup
              title="Portrayed By"
              value={undefined}
            />
          </dl>
          
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <h2>
            Gwen
          </h2>
          <dl
            className="dl-horizontal"
          >
            <PropertyGroup
              title="Gender"
              value={undefined}
            />
            <PropertyGroup
              title="Culture"
              value={undefined}
            />
            <PropertyGroup
              title="Born"
            >
              <LifeEventField
                date={undefined}
                location={undefined}
              />
            </PropertyGroup>
            <PropertyGroup
              title="Died"
            >
              <LifeEventField
                date={undefined}
                location={undefined}
              />
            </PropertyGroup>
            <PropertyGroup
              title="Aliases"
              value={undefined}
            />
            <PropertyGroup
              title="Titles"
              value={undefined}
            />
            <PropertyGroup
              title="Portrayed By"
              value={undefined}
            />
          </dl>
          
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <CharacterDetails
    character={
      Object {
        "name": "Gwen",
      }
    }
  />,
}
`;
